src/protobuf-net.Reflection/CodeGenerator.cs

```diff
@@ -30,7 +30,7 @@ public abstract class CodeGenerator
         /// <summary>
         /// Execute the code generator against a FileDescriptorSet, yielding a sequence of files
         /// </summary>
-        public abstract IEnumerable<CodeFile> Generate(FileDescriptorSet set, NameNormalizer normalizer = null, Dictionary<string, string> options = null);
+        public abstract IEnumerable<CodeFile> Generate(FileDescriptorSet set, NameNormalizer normalizer = null, Dictionary<string, string> options = null, Dictionary<string, string> customNamesConversion = null);
```

Comment:
breaking change; needs to be binary compatible - add as new overload, and change the *old* version so that all params are required. The alternative is we say "to hell with it", break binary compatibilty, and increment the "major". I'd rather not do that.

Reply:
How about add a new virtual method `public virtual IEnumerable<CodeFile> Generate(FileDescriptorSet set, NameNormalizer normalizer = null, Dictionary<string, string> options = null, Dictionary<string, string> customNamesConversion = null)` for override ?